ql/src/java/org/apache/hadoop/hive/ql/exec/tez/KillTriggerActionHandler.java

```diff
@@ -38,28 +40,27 @@ public class KillTriggerActionHandler implements TriggerActionHandler<TezSession
   @Override
   public void applyAction(final Map<TezSessionState, Trigger> queriesViolated) {
     for (Map.Entry<TezSessionState, Trigger> entry : queriesViolated.entrySet()) {
-      switch (entry.getValue().getAction().getType()) {
-      case KILL_QUERY:
-        TezSessionState sessionState = entry.getKey();
-        String queryId = sessionState.getWmContext().getQueryId();
-        try {
-          UserGroupInformation ugi = UserGroupInformation.getCurrentUser();
-          SessionState ss = new SessionState(new HiveConf(), ugi.getShortUserName());
-          ss.setIsHiveServerQuery(true);
-          SessionState.start(ss);
-          KillQuery killQuery = sessionState.getKillQuery();
-          // if kill query is null then session might have been released to pool or closed already
-          if (killQuery != null) {
-            sessionState.getKillQuery().killQuery(queryId, entry.getValue().getViolationMsg(),
-                      sessionState.getConf());
-          }
-        } catch (HiveException|IOException e) {
-          LOG.warn("Unable to kill query {} for trigger violation", queryId);
+        if (Objects.requireNonNull(entry.getValue().getAction().getType()) == Action.Type.KILL_QUERY) {
+            TezSessionState sessionState = entry.getKey();
+            String queryId = sessionState.getWmContext().getQueryId();
+            try {
+                UserGroupInformation ugi = UserGroupInformation.getCurrentUser();
+                try (SessionState ss = new SessionState(new HiveConf(), ugi.getShortUserName())) {
```

Comment:
similar to other comments, do we need to create and start new session per tez session entry?